src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceCredential.cs

```diff
@@ -67,7 +67,7 @@ public string Password
                 }
                 else
                 {
-                    return PasswordText;
+                    return PasswordText!;
```

Comment:
I think this is one of the places I was getting frustrated with the design/implementation, so I just suppressed the issue.

The statis `FromUserInput` throws if `password` is null. So, if an instance is created that way, `Password` and `PasswordText` can't be null.

However, the class constructor doesn't throw if password is null. So, maybe a null is allowed?  But after I finished annotating everything, there weren't any calls to the constructor where the compiler detected that a null value might be used.

So, in the end I just suppressed it, because that's how the implementation appears to work, even if the way it's written confuses the compiler. If you do a null check in an if statement, even if the symbol is not marked as nullable, the compiler will treat it as nullable. So, the `if (PasswordText != null && otherCondition)` tricks the compiler into thinking that `PasswordText` might be null when the `if` expression is false, even though `PasswordText` is not marked as nullable.

In theory it should be safe to remove the `PasswordText != null` part of the expression, but my goal is primarily to add nullable annotations, and I want to minimize risk of adding bugs.

Reply:
Okay